recipes/grpc/all/conanfile.py

```diff
@@ -61,10 +61,9 @@ def requirements(self):
         self.requires('re2/20210601')
 
     def build_requirements(self):
-        self.build_requires('protobuf/3.17.1')
-
-        #when cross compiling we need pre compiled grpc plugins for protoc
+        # when cross compiling we need pre compiled grpc plugins for protoc
         if hasattr(self, "settings_build") and tools.cross_building(self):
+            self.build_requires('protobuf/3.17.1')
             self.build_requires('grpc/{}'.format(self.version))
```

Comment:
```suggestion
        if hasattr(self, "settings_build"):
            self.build_requires('protobuf/3.17.1')
            if tools.cross_building(self)
                self.build_requires('grpc/{}'.format(self.version))
```

The issue is not cross-build, but 1 profile which can't override build requirements.

Reply:
Thank you for clarifying the usage. I have updated the pull request.